assets/constructor_macOS_common.patch

```diff
@@ -1,101 +0,0 @@
-diff -Naur constructor-orig/osx/post_extract.sh constructor-patched/osx/post_extract.sh
---- constructor-orig/osx/post_extract.sh	2022-02-09 14:17:06.000000000 +0100
-+++ constructor-patched/osx/post_extract.sh	2022-03-19 20:28:13.000000000 +0100
-@@ -10,7 +10,8 @@
-
- unset DYLD_LIBRARY_PATH
-
--PREFIX="$2/__NAME_LOWER__"
-+# hard-code prefix, as we cannot specify it via construct.yaml
-+PREFIX="$2/.mne-python"
- PREFIX=$(cd "$PREFIX"; pwd)
- export PREFIX
- echo "PREFIX=$PREFIX"
-@@ -24,7 +25,7 @@
-
- # Extract the conda packages but avoiding the overwriting of the
- # custom metadata we have already put in place
--notify "Preparing packages..."
-+logger -p 'install.info' "Preparing packages..."
- "$CONDA_EXEC" constructor --prefix "$PREFIX" --extract-conda-pkgs
- if (( $? )); then
-     echo "ERROR: could not extract the conda packages"
-@@ -32,7 +33,7 @@
- fi
-
- # Perform the conda install
--notify "Installing packages. This might take a few minutes."
-+logger -p 'install.info' "Installing packages. This might take a few minutes."
- CONDA_SAFETY_CHECKS=disabled \
- CONDA_EXTRA_SAFETY_CHECKS=no \
- CONDA_CHANNELS=__CHANNELS__ \
-@@ -65,9 +66,9 @@
-
-
- if [ -n '__SHORTCUTS__' ]; then
--    notify "Done! You can open __NAME__ via ~/Applications."
-+    logger -p 'install.info' "Done! You can find __NAME__ in the Applications folder."
- else
--    notify "Done! Installation is available in $PREFIX."
-+    logger -p 'install.info' "Done! Installation is available in $PREFIX."
- fi
-
- echo "installation to $PREFIX finished."
-diff -Naur constructor-orig/osxpkg.py constructor-patched/osxpkg.py
---- constructor-orig/osxpkg.py	2022-02-09 14:17:06.000000000 +0100
-+++ constructor-patched/osxpkg.py	2022-03-19 20:15:12.000000000 +0100
-@@ -56,7 +56,7 @@
-     root = tree.getroot()
-
-     title = ET.Element('title')
--    title.text = info['name']
-+    title.text = f"{info['name']} {info['version']}"
-     root.append(title)
-
-     license = ET.Element('license', file=info.get('license_file',
-@@ -153,8 +153,9 @@
-     # See below for an explanation of the consequences of this
-     # customLocation value.
-     for options in root.findall('options'):
--        options.set('customize', 'allow')
--        options.set('customLocation', '/')
-+        options.set('customize', 'never')
-+        options.set('customLocation', '/Applications/MNE-Python')
-+        options.set('rootVolumeOnly', 'true')
```

Comment:
@jaimergp I guess we should somehow expose these options in constructor

Reply:
When I looked into this, literature showed that root-only installations tend to be buggy across different macOS versions (a few SO questions plus a couple blog posts), hence why we default to user-installs with an option to override if needed. 

xref: https://stackoverflow.com/questions/11487596/making-macos-installer-packages-which-are-developer-id-ready/11487658#11487658

Honestly, I am sure it's technically feasible, but given how many installations are performed globally using this infra (Anaconda alone must add up to A LOT), I wouldn't want to support questions from all those edge cases.

Reply:
Okay, it sounds like we *shouldn't* implement this and instead recommend / follow this better practice of using user-installs by default!